content/technotes/2020-02-26-tic-ghactions.Rmd

```diff
@@ -0,0 +1,170 @@
+
+However, [Travis CI](https://travis-ci.com/) has it limits and therefore people started to try out other providers such as [Appveyor CI](https://www.appveyor.com/) or [Circle CI](https://circleci.com/).
+Besides having to start from scratch, all providers come with their own YAML keys and ways to set up deployment permissions.
+
+With the aim to simplify these processes for users, the tic package was created.
+It is aimed to provide a CI-agnostic way to check R packages.
+But not just R packages! 
+It also aims to simplify the creation of blogs, websites, [drat](https://github.com/eddelbuettel/drat) repositories and other things related to CI activities.
+
+By providing CI YAML templates which forward their actions to a centralized R script (`tic.R`), the same setup will trigger identical actions on different CI systems.
+The user does not need to worry about setting up the YAML file for a specific provider but just get familiarized with the "macro" approach of tic.
+[Macros](https://docs.ropensci.org/tic/articles/tic.html#macros) are sequences of steps for popular actions often requested by people in the R community for R related projects.
+
+Furthermore tic aims to simplify the tedious process of setting up deployment permissions for a repo.
+("Deployment" means that a CI run is allowed to perform a `git` push to the repository.)
+This is achieved by its helper packages such as circle, ghactions or travis which perform API calls to the respective CI providers.
+
+### GitHub Actions - the new kid on the block
+
+GitHub Actions (beta) is around since about one year.
+The official release happened in [November 2019](https://siliconangle.com/2019/11/13/github-universe-announcements-bring-bevy-updates-developers/).
+The hype after the launch was huge: People are used to expect properly developed products from GitHub and having an integrated CI solution without the overhead of connecting to a third party provider sounded promising.
+
+Indeed, GitHub Actions fulfilled this promise: It is way easier to get started with CI now: 
+
+- No extra account needs to be created for a third-pary service
+- No restrictions on the available build platforms
+- No issues related to webhooks of third-party providers
+- CI settings configurable via the repository settings
+
+However, R was again not among the languages with native support/examples, so it was not easy to get started.
+After a first shot from [Max Held](https://github.com/maxheld83) via the [ghactions](https://github.com/maxheld83/ghactions) package, [Jim Hester](https://github.com/jimhester) sat down again and created the [setup-r](https://github.com/r-lib/actions/tree/master/setup-r) actions which installs R on all avaialable platforms with the option to specify different R versions.
+
+While this action is already usable, its still in its early days and will mature over time.
```

Comment:
```suggestion
While the action package is already usable, it's still in its early days and will mature over time.
```

Reply:
It's not really an R package (even though it has a DESC).
I would like to refer to it as a GitHub action rather than an R package (rephrased manually).